pkg/api/types.go

```diff
@@ -1410,6 +1412,55 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
```

Comment:
I think the intention was to make it an API resource.

But I'm not sure I agree about adding a selector. I don't think this is an object normal users should be creating or thinking about. IMO it should be created by the controller managing a Deployment and the controller managing a Job (we'd extend the spec for those to include specifying a disruption budget), which should also be responsible for maintaining the back-pointers from the pods they manage to the DisruptionBudget object. Given the back-pointers, I don't think a selector on DisruptionBudget is needed. This trades off some flexibility but I think there are significant usability benefits to saying a DisruptionBudget maps 1:1 with those controllers. (In this model there is also just one DisruptionBudget per pod.)

A slightly more complicated model is to associate DisruptionBudget with Service (instead of Deployment or Job). In that case a pod can be controlled by more than one DisruptionBudget. But you still wouldn't need a selector in the DisruptionBudget.